Framework/PythonInterface/mantid/api/src/Exports/IFunction.cpp

```diff
@@ -35,6 +36,32 @@ PyObject *getCategories(IFunction &self) {
   return registered;
 }
 
+/// An object for constructing a shared_ptr that won't ever delete its pointee
+class NoDeleting {
+public:
+  /// Does nothing
+  void operator()(void *) {}
+  /// Does nothing
+  void operator()(const void *) {}
+};
+
+//------------------------------------------------------------------------------------------------------
+/**
+* Something that makes IFunction return to python a composite function
+* for Product function, Convolution or
+* any similar superclass of composite function.
+* @param self :: Enables it to be called as a member function on the
+* IFunction class
+*/
+Mantid::API::CompositeFunction_sptr castToCompositeFunction(IFunction &self) {
```

Comment:
My feeble understanding is that a raw pointer will not prevent the python garbage collector to delete the pointee once the associated python object reference count reaches zero. If we use a raw pointer, will `self.fun` in Line 253 of `FunctionWrapper.py` be deleted once variable `comp` goes out of scope?